lib/stdlib/js/sys_stdin.js

```diff
@@ -0,0 +1,3 @@
+function set_stdin_wrapper(vm){
+    return vm.exec('import sys,StringIO\nclass fifoFile(object):\n    buffer=StringIO.StringIO()\n    def close(self): pass\n    def read(self, *args, **kw):\n        self.buffer.seek(0)\n        line=self.buffer.read(*args, **kw)\n        self.buffer.buf=self.buffer.buf[len(line):]\n        self.buffer.read()\n        return line\n    def readline(self):\n        self.buffer.seek(0)\n        line = self.buffer.readline()\n        self.buffer.buf=self.buffer.buf[len(line):]\n        self.buffer.read()\n        return line\n\nsys.stdin = fifoFile();stdinwrapper=js.Function(sys.stdin.buffer.write);js.globals["vm"].stdin=stdinwrapper;import os;os.fileNumbers[0]=sys.stdin\n');
```

Comment:
It's times like this I really miss python's triple-quoted strings in javascript...